.github/workflows/ci_cmake.yml

```diff
@@ -0,0 +1,123 @@
+      - 'build/tools/travis-ci.sh'
+      - 'distrib/**'
+      - 'docs/**'
+      - 'interface/**'
+      - 'include/msvc/**'
+      - 'include/wx/msw/**'
+      - 'locale/**'
+      - 'src/msw/**'
+      - '*.md'
+      - '*.yml'
+      - 'wxwidgets.props'
+
+jobs:
+  build:
+    runs-on: ${{ matrix.runner }}
+    name: ${{ matrix.name }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: Ubuntu 18.04 wxGTK 3
+            runner: ubuntu-18.04
+            cmake_generator: Unix Makefiles
+          - name: macOS 10.15 wxOSX
+            runner: macos-10.15
+            cmake_generator: Xcode
+            cmake_defines: -DCMAKE_CXX_STANDARD=11
+          - name: macOS 10.15 wxIOS
+            runner: macos-10.15
+            cmake_generator: Xcode
+            cmake_defines: -DCMAKE_SYSTEM_NAME=iOS -DCMAKE_FIND_ROOT_PATH=/usr/local -DCMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_ALLOWED=NO
+            cmake_samples: OFF
+            cmake_tests: OFF
+
+    env:
+      wxGTK_VERSION: ${{ matrix.gtk_version && matrix.gtk_version || 3 }}
+      wxCMAKE_GENERATOR: ${{ matrix.cmake_generator }}
+      wxCMAKE_DEFINES: ${{ matrix.cmake_defines }}
+      wxCMAKE_TESTS: ${{ matrix.cmake_tests }}
+      wxCMAKE_SAMPLES: ${{ matrix.cmake_samples }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+
+      - name: Set environment variables
+        run: |
+          case `uname` in
+            Linux)
+                wxPROC_COUNT=`nproc`
+                ;;
+
+            Darwin)
+                wxPROC_COUNT=`sysctl -n hw.ncpu`
+                ;;
+
+            *)
+                echo "*** Unknown platform: `uname` ***"
+                wxPROC_COUNT=0
+                ;;
+          esac
+
+          ((wxPROC_COUNT++))
+          if [ "$wxCMAKE_GENERATOR" == "Xcode" ]; then
+              wxBUILD_ARGS="-jobs $wxPROC_COUNT -quiet"
+          else
+              wxBUILD_ARGS=-j$wxPROC_COUNT
+          fi
+          echo wxBUILD_ARGS=$wxBUILD_ARGS >> $GITHUB_ENV
+
+          if [ -z $wxCMAKE_TESTS ]; then wxCMAKE_TESTS=CONSOLE_ONLY; fi
+          echo wxCMAKE_TESTS=$wxCMAKE_TESTS >> $GITHUB_ENV
+          if [ -z $wxCMAKE_SAMPLES ]; then wxCMAKE_SAMPLES=SOME; fi
+          echo wxCMAKE_SAMPLES=$wxCMAKE_SAMPLES >> $GITHUB_ENV
+
+          # Setting this variable suppresses "Error retrieving accessibility bus address"
+          # messages from WebKit tests that we're not interested in.
+          echo NO_AT_BRIDGE=1 >> $GITHUB_ENV
+
+      - name: Before install
+        run: |
+          ./build/tools/before_install.sh
+
+      - name: Configuring
+        run: |
+          cmake --version
+          mkdir build_cmake
+          pushd build_cmake
+          cmake -G "$wxCMAKE_GENERATOR" $wxCMAKE_DEFINES -D wxBUILD_SAMPLES=$wxCMAKE_SAMPLES -D wxBUILD_TESTS=$wxCMAKE_TESTS ..
+
+      - name: Building
+        run: |
+          pushd build_cmake
```

Comment:
Could also use `working-directory: build_cmake`, I used it for some steps because I thought it was a bit more clear.

Reply:
added